src/Client.js

```diff
@@ -969,7 +1025,88 @@ class Client extends EventEmitter {
             return await window.WWebJS.getChat(chatId);
         }, chatId);
 
-        return ChatFactory.create(this, chat);
+        return chat
+            ? ChatFactory.create(this, chat)
+            : chat;
+    }
+
+    /**
+     * @typedef {Object} GetChannelOptions
+     * @property {boolean} [getMetadata = false] If true, gets the {@link ChannelMetadata}, false by default
+     */
+
+    /**
+     * @typedef {Object} ChannelMetadata
+     * @property {string} id The channel ID in a format: 'XXXXXXXXXX@newsletter'
+     * @property {number} createdAtTs The timestamp the channel was created at
+     * @property {Object} titleMetadata
+     * @property {string} titleMetadata.title The channel title
+     * @property {number} titleMetadata.updatedAtTs The timestamp the title was updated at
+     * @property {Object} descriptionMetadata
+     * @property {string} descriptionMetadata.description The channel description
+     * @property {string} descriptionMetadata.updatedAtTs The timestamp the description was updated at
+     * @property {string} inviteLink The channel invite link in a format: https://whatsapp.com/channel/INVITE_CODE
+     * @property {string} membershipType The membership type of a current user in a channel (viewer/guest/subscriber/admin/owner)
+     * @property {string} stateType The channel state type (active/suspended/geosuspended)
+     * @property {string} pictureUrl The channel profile picture url if set
+     * @property {number} subscribersCount The number of channel subscribers
+     * @property {boolean} isVerified Indicates if a channel is verified
+     */
+
+    /**
+     * Gets a {@link Channel} object or a {@link ChannelMetadata} by its ID
+     * @param {string} channelId 
+     * @param {GetChannelOptions} options
+     * @returns {Promise<ChannelMetadata|Channel>}
+     */
+    async getChannelById(channelId, options = {}) {
+        const result = await this.pupPage.evaluate(async (channelId, options) => {
+            if (options.getMetadata) {
+                try {
+                    return await window.WWebJS.getChannelMetadata(channelId, { getById: true });
```

Comment:
Why not returnig the metadata as part as the channel object?

Reply:
@shirser121 

> Why not returnig the metadata as part as the channel object?

The `channelMetadata` is already [returned](https://github.com/alechkos/whatsapp-web.js/blob/e485655ae3e4403a3d0622ed73e2b80a778f41d1/src/structures/Channel.js#L26) as a part of a `Channel` object.
Here the 'metadata' name stands for like a channel preview that you can get, not a real metadata that is returned as a part of a `Channel` object